src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -36,4 +84,73 @@ protected function getExtension()
     {
         return 'yml';
     }
+
+    private function expandToTree(array $messages)
+    {
+        $tree = array();
+
+        foreach ($messages as $id => $value) {
+            $this->expandElement($tree, explode('.', $id), $value);
```

Comment:
I think as_tree dumping should not be used in this case.
Tree is usefull if you work with your translations by abstract constant string ID, not by their real writing.
So, if you use something like `security.resetting.login` - use the tree dumping, if as in example above - linear.